### PR TITLE
various cgroups mini fixes

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/cgroup-util.c [hunk 1]
```diff
@@ -2303,8 +2303,10 @@ int cg_mask_to_string(CGroupMask mask, char **ret) {
         return 0;
```
Comment (keszybz) [inline src/basic/cgroup-util.c:2315]: (comment for the patch description)

"[no one] cares for smth" means that every dislikes / has a problem with smth. I think you mean "care about" which is neutral.

File: src/basic/cgroup-util.h [hunk 2]
```diff
@@ -49,6 +49,16 @@ typedef enum CGroupMask {
         CGROUP_MASK_PIDS = CGROUP_CONTROLLER_TO_MASK(CGROUP_CONTROLLER_PIDS),
         CGROUP_MASK_BPF_FIREWALL = CGROUP_CONTROLLER_TO_MASK(CGROUP_CONTROLLER_BPF_FIREWALL),
         CGROUP_MASK_BPF_DEVICES = CGROUP_CONTROLLER_TO_MASK(CGROUP_CONTROLLER_BPF_DEVICES),
+
+        /* All real cgroupv1 controllers */
+        CGROUP_MASK_V1 = CGROUP_MASK_CPU|CGROUP_MASK_CPUACCT|CGROUP_MASK_BLKIO|CGROUP_MASK_MEMORY|CGROUP_MASK_DEVICES|CGROUP_MASK_PIDS,
```
Comment (keszybz) [inline src/basic/cgroup-util.h:54]: Look, cgroupsv1 has so many more features and is so much better!

(just joking)

File: src/basic/cgroup-util.c [hunk 3]
```diff
@@ -2168,6 +2171,9 @@ int cg_attach_everywhere(CGroupMask supported, const char *path, pid_t pid, cg_m
                 CGroupMask bit = CGROUP_CONTROLLER_TO_MASK(c);
                 const char *p = NULL;
 
+                if (!FLAGS_SET(CGROUP_MASK_V1, bit))
+                        continue;
+
                 if (!(supported & bit))
```
Comment (keszybz) [inline src/basic/cgroup-util.c]: This and places below could be switched over to `FLAGS_SET` too, maybe later.

File: src/basic/cgroup-util.c [hunk 4]
```diff
@@ -2377,16 +2377,14 @@ int cg_mask_supported(CGroupMask *ret) {
...
                  * everything else off. */
-                mask &= CGROUP_MASK_CPU | CGROUP_MASK_MEMORY | CGROUP_MASK_IO | CGROUP_MASK_PIDS;
+                mask &= CGROUP_MASK_V2;
 
         } else {
                 CGroupController c;
 
-                /* In the legacy hierarchy, we check whether which
-                 * hierarchies are mounted. */
+                /* In the legacy hierarchy, we check whether which hierarchies are mounted. */
```
Comment (keszybz) [inline src/basic/cgroup-util.c]: "whether" should go.

File: src/basic/cgroup-util.c [hunk 5]
```diff
@@ -2396,6 +2394,8 @@ int cg_mask_supported(CGroupMask *ret) {
                         if (controller_is_accessible(n) >= 0)
```
Comment (keszybz) [inline src/basic/cgroup-util.c:2398]: Maybe shortcut the check if `c` is not in `CGROUP_MASK_V1`?

File: src/basic/cgroup-util.c [hunk 6]
```diff
@@ -129,10 +129,12 @@ bool cg_ns_supported(void) {
         if (enabled >= 0)
                 return enabled;
 
-        if (access("/proc/self/ns/cgroup", F_OK) == 0)
-                enabled = 1;
-        else
-                enabled = 0;
+        if (access("/proc/self/ns/cgroup", F_OK) < 0) {
+                if (errno != ENOENT)
+                        log_debug_errno(errno, "Failed to check whether /proc/self/ns/cgroup is available, ignoring: %m");
```
Comment (keszybz) [inline src/basic/cgroup-util.c]: "ignoring" → "assuming no" ?

File: src/basic/cgroup-util.c (hunk 3 above)
Comment (poettering) [reply src/basic/cgroup-util.c]: will add a trivial commit on top that adds this

**Comment by poettering** [comment]
force pushed a new version that fixes all issues mentioned. ptal

**Comment by poettering** [comment]
ah, wait, given that the final changes were so trivial, let's also set the green label on this one.

**Pull request merged**